In [1]:
import pandas as pd
import numpy as np
import tensorflow as tf
import seaborn as sns 
import matplotlib.pyplot as plt
%matplotlib inline 
sns.set(color_codes=True)

Tablica koja prikazuje prosječni score svake osobe za svaku vještinu i koju strategiju je ta osoba odabrala.

In [2]:
df = pd.read_excel("SEVER_2.xlsx")
df.iloc[:,0] = df.iloc[:,0]/10
df.iloc[:,1] = df.iloc[:,1]/10
df.iloc[:,2] = df.iloc[:,2]/10
df.head(5)

Vjestina aktivnog slusanja  Vjestina umrezavanja  \
0                         4.6                   4.0   
1                         4.2                   3.6   
2                         4.8                   4.2   
3                         4.8                   4.8   
4                         4.0                   5.0   

   Vjestina utjecaja na druge  Tvrdo pregovaranje  Pozicijsko pregovaranje  \
0                         3.6                   0                        0   
1                         5.0                   0                        0   
2                         4.4                   0                        1   
3                         4.2                   0                        0   
4                         4.6                   1                        0   

   Meko pregovaranje  Fiktivno pregovaranje  Principijelno pregovaranje  
0                  0                      0                           1  
1                  0                      0                           1  
2                  0                      0                           0  
3                  0                      0                           1  
4                  0                      0                           0

In [8]:
# Create an empty list
filter_arr = []
# go through each element in arr
for i in range(0, 30):
  if 1 == df.iloc[i,7]:
    filter_arr.append(True)
  else:
    filter_arr.append(False)

In [17]:
vas_princip = df.iloc[:,0][filter_arr].to_numpy()

In [26]:
filter_arr_1 = np.invert(filter_arr)

In [27]:
filter_arr_1

array([False, False,  True, False,  True, False, False, False, False,
        True, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False])

In [28]:
vas_neprincip = df.iloc[:,0][filter_arr_1].to_numpy()

In [30]:
np.mean(vas_princip)

4.214814814814814

Prosjek za vještinu aktivnog slušanja za osobe koje su odabrale principijelno pregovaranje je 4.21481.

In [31]:
np.mean(vas_neprincip)

4.266666666666667

Prosjek za vještinu aktivnog slušanja za osobe koje su odabrale druge strategije je 4.26667. 

Dakle nema značajne razlike u prosjecima populacija za tu vještinu.

In [32]:
vu_princip = df.iloc[:,1][filter_arr].to_numpy()

In [33]:
vu_neprincip = df.iloc[:,1][filter_arr_1].to_numpy()

In [37]:
np.mean(vu_princip)

4.148148148148148

Prosjek za vjestinu umrezavanja za osobe koje su izabrale principijelno pregovaranje je 4.148148..

In [38]:
np.mean(vu_neprincip)

4.333333333333333

Prosjek za vjestinu umrezavanja za osobe koje nisu izabrale principijelno pregovaranje je 4.33333....

Nema neke ocite razlike izmedju prosjeka.

In [39]:
vud_princip = df.iloc[:,2][filter_arr].to_numpy()
vud_neprincip = df.iloc[:,2][filter_arr_1].to_numpy()

In [40]:
np.mean(vud_princip)

4.288888888888889

Prosjek za vjestinu utjecaja na druge za osobe koje su birale principijelno pregovaranje je 4.288888...

In [41]:
np.mean(vud_neprincip)

4.3999999999999995

Prosjek za vjestinu utjecaja na druge za osobe koje nisu birale principijelno pregovaranje je 4.399999...

Nema neke znacajne razlike.

## Whitney Mann U Test

Whitney Mann U test ima nul-hipotezu H0: distribucija jedne populacije je jednaka distribuciji druge populacije.

Prvo ćemo usporediti distribuciju vjestine aktivnog slusanja izmedju osoba koje su odabrale principijelno pregovaranje i onih koji nisu.

In [42]:
# Uzorak osoba za vjestinu aktivnog slusanja koje su odabrale principijelno pregovaranje
vas_princip

array([4.6, 4.2, 4.8, 4. , 3.4, 4.4, 3.6, 4.6, 4. , 5. , 3.6, 3.8, 4. ,
       3.6, 4.4, 4.6, 4.4, 4.8, 4.4, 4.4, 3.6, 4.4, 4.2, 4. , 4. , 4.6,
       4.4])

In [43]:
# Uzorak osoba za vjestinu aktivnog slusanja koje nisu odabrale principijelno pregovaranje
vas_neprincip

array([4.8, 4. , 4. ])

In [44]:
import scipy
scipy.stats.mannwhitneyu(vas_princip, vas_neprincip)

MannwhitneyuResult(statistic=39.0, pvalue=0.9440278624662525)

Jako velika p-vrijednost (cca 94%), dakle ne odbacujemo H0 hipotezu, odnosno ne odbacujemo hipotezu koja kaze da su distribucije tih dviju populacija jednake.

Sada na isti nacin ispitamo za vjestinu umrezavanja

H0: vjestine umrezavanja za one ljude koji su birali principijelno pregovaranje imaju jednaku distribuciju kao i vjestine umrezavanja onih ljudi koji su birali druge strategije

In [45]:
import scipy
scipy.stats.mannwhitneyu(vu_princip, vu_neprincip)

MannwhitneyuResult(statistic=34.5, pvalue=0.7008867365309154)

Jako velika p-vrijednost (cca 70%), dakle ne odbacujemo H0 hipotezu.

I na isti nacin ispitamo i za trecu vjestinu.

In [46]:
import scipy
scipy.stats.mannwhitneyu(vud_princip, vud_neprincip)

MannwhitneyuResult(statistic=35.5, pvalue=0.7527350317508261)

Opet velika p-vrijednost (cca 75%), dakle ne odbacujemo H0.